cmd/worker_cli/worker_cli.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/MinhNghiaD/jobworker/api/client"
+	"github.com/MinhNghiaD/jobworker/api/worker/proto"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/status"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	address = kingpin.Flag("a", "server address").Default("127.0.0.1:7777").String()
+	// TODO add flags for certificate, private key and server CAs
+
+	// Start subcommand and its flags
+	start     = kingpin.Command("start", "Start a job on worker service.")
+	startCmd  = start.Flag("cmd", "command to be executed").Default("").String()
+	startArgs = start.Arg("args", "arguments of the command").Strings()
+
+	// Stop subcommand and its flags
+	stop        = kingpin.Command("stop", "Stop a job on worker service.")
+	stopForce   = stop.Flag("force", "force job to terminate immediately").Default("false").Bool()
+	stoppingJob = stop.Flag("job", "job id").Default("").String()
+
+	// Query subcommand and its flags
+	query      = kingpin.Command("query", "Query status of a job on worker service.")
+	queriedJob = query.Flag("job", "job id").Default("").String()
+
+	// TODO: add Stream subcommand
+)
+
+func main() {
+	subCommand := kingpin.Parse()
+
+	cli, err := client.New(*address)
+	if err != nil {
+		logrus.Fatalf("Fail to init client %s", err)
+	}
+	defer cli.Close()
+
+	switch subCommand {
+	case start.FullCommand():
+		startJob(cli, *startCmd, *startArgs)
+	case stop.FullCommand():
+		stopJob(cli, *stoppingJob, *stopForce)
+	case query.FullCommand():
+		queryJob(cli, *queriedJob)
```

Comment:
Why not have these functions return an error then you can consolidate `reportError` and error with exit code logic here?

Reply:
My bad.  I have committed a patch